libr/anal/dwarf_process.c

```diff
@@ -1062,6 +1068,9 @@ static VariableLocation *parse_dwarf_location (Context *ctx, const RBinDwarfAttr
 			reg_name = get_dwarf_reg_name (ctx->anal->cpu, reg_num, &kind, ctx->anal->bits);
 		} break;
 		case DW_OP_bregx: {
+			if (i == block.length - 1) {
+				return NULL;
+			}
 			/* 2 operands, reg_number, offset*/
 			/* I need to find binaries that uses this so I can test it out*/
 			const ut8 *buffer = &block.data[++i];
```

Comment:
I think there should be an additional if check before the next uleb128() as there are two in this case.

Reply:
I'm eoutoftime right now -- will check `r_uleb128()` tomorrow.

Reply:
> I think there should be an additional if check before the next uleb128() as there are two in this case.

Okay 604f3715007fe241bf00933cc6a14fd1ca5572a6 ... I think the first check is unnecessary but it's harmless and its psychological impact should not be underestimated.

Reply:
I think both checks are useful here outside of the psychological impact, one for each `leb128` as they don't do bound checking on the first byte they read.

Reply:
I agree, could you approve the pr?

Reply:
Actually `r_uleb128()` does check whether the length is 0 but users shouldn't need to know that and there is a need to `return NULL` anyway.

Reply:
Oh true, might be cool to have same interfaces for boh `leb` and `uleb` in the future and have them to make the same checks :D

Reply:
Have to verify correctness of each change though :dizzy_face: